text/0066-separate-context-propagation.md

```diff
@@ -0,0 +1,559 @@
+# Proposal: Separate Layer for Context Propagation
+
+* [OpenTelemetry layered architecture](#openTelemetry-layered-architecture)
+  * [Aspects](#Aspects)
+    * [Observability API](#Observability-API)
+    * [Baggage API](#Baggage-API)
+  * [Context Propagation](#Context-Propagation)
+    * [Context API](#Context-API)
+    * [Propagation API](#Propagation-API)
+* [Prototypes](#Prototypes)
+* [Examples](#Examples)
+* [Internal details](#Internal-details)
+* [FAQ](#faq)
+
+Refactor OpenTelemetry into a set of separate cross-cutting concerns which 
+operate on a shared context propagation mechanism.
+
+## Motivation
+
+This RFC addresses the following topics:
+
+**Separatation of concerns**  
+* Remove the Tracer dependency from context propagation mechanisms.
+* Handle user data (Baggage) and observability data (SpanContext, Correlations) 
+  separately.  
+
+**Extensibility**
+* Allow developers to create new applications for context propagation. For 
+  example: A/B testing, encrypted or authenticated data, and new, experimental 
+  forms of observability.
+
+## Explanation
+
+# OpenTelemetry layered architecture
+
+The design of OpenTelemetry is based on the priciples of aspect-oriented 
+programming, adopted to the needs of distributed systems.
+
+OpenTelemetry is separated into two layers: **aspects** which crosscut and 
```

Comment:
I was confused by the wording of this sentence/paragraph on my first read through. The way the beginning of the sentence was worded, I expected the second layer to immediately follow **aspects**. This left me with the question “Is aspect a layer or is each layer an aspect,” which required further reading beyond this paragraph to understand.

Suggest:
> OpenTelemetry is separated into two layers: **aspects** and **context propagation**. Aspects crosscut and intertwine with a program’s functional concerns, and cannot be encapsulated. Multiple aspects - including the tracing and baggage systems provided by OpenTelemetry - then share the same underlying context propagation system, which allows these cross-cutting concerns to store and access their data across the lifespan of a distributed transaction.